06-building-a-program-start-to-finish/writing-tests/output/invertedIndexSpec.js

```diff
@@ -0,0 +1,161 @@
+const books = require('../jasmine/assets/correct.json');
+const wrongdata = require('../jasmine/assets/wrongFormat.json');
+const emptyfile = require('../jasmine/assets/emptyfile.json');
+const smallfile = require('../jasmine/assets/smallcorrectfile.json');
+const uniqueTermsBook = require('../jasmine/assets/uniqueTermsBook.json');
+
+describe('InvertedIndex Class', () => {
+  beforeAll(() => {
+    this.invertedIndex = new InvertedIndex();
+  });
+
+  describe('Constructor', () => {
+    it('can create inverted index instance', () => {
+      expect(typeof this.invertedIndex)
+        .toEqual('object');
+      expect(this.invertedIndex instanceof InvertedIndex)
+        .toBe(true);
+    });
+
+    it('has an indexes object to hold all indexes', () => {
+      expect(typeof this.invertedIndex.tableObj)
+        .toEqual('object');
+    });
+  });
+
+  describe('tokenize', () => {
+    it('checks the tokenize function', () => {
+      const input = 'jed is a boy'
+      const output = ['jed', 'is', 'a', 'boy'];
+      expect(this.invertedIndex.tokenize(input))
+        .toEqual(output);
+    });
+
+    it('sanitizes the input in the tokenize function', () => {
+      const input = 'tracy!!! is !@#$%^&*not -invited. &to my (wedding)'
+      const output = ['tracy', 'is', 'not', 'invited', 'to', 'my', 'wedding'];
+      expect(this.invertedIndex.tokenize(input))
+        .toEqual(output);
+    });
+  });
+
+  describe('getTextFromJsonObj', () => {
+    it('should return an array of words', () => {
+      expect(this.invertedIndex.getTextFromJsonObj(smallfile))
+        .toEqual(['alice', 'falls', 'into', 'a', 'rabbit']);
+    });
+
+    it('filters out symbols', () => {
+      expect(this.invertedIndex.getTextFromJsonObj(smallfile))
+        .toEqual(['alice', 'falls', 'into', 'a', 'rabbit']);
+    });
+  });
+
+  describe('uniqueWords', () => {
+    it('checks the uniqueWords function', () => {
```

Comment:
`checks the uniqueWords function`. Same as above.

Please ensure your specs actually communicate what you're testing for BDD Style.